docs/api.md

```diff
@@ -611,6 +612,93 @@ endpoints without notice.
     issued by external issuers. These statistics only include codes issued by
     the API where an `externalIssuer` field was provided.
 
+# User report webhooks
+
+You can use your own gateway to dispatch SMS messages for user reports. When a
+user completes a self report, the verification server will send a request with
+the compiled SMS message and destination phone number to your server.
+
+- Must be a **publicly-accessible** endpoint
+- Must be **unauthenticated**
+- Must be **secured via TLS** (https)
+- Must accept a **POST** request
+- Must parse the response as JSON
+
+The request body will be identical to the [API /issue response](#apiissue). Your
+server should parse the JSON body and extract the `generatedSMS` field.
+
+Before accepting the request, your server **MUST** validate the integrity of the
+request. All messages from the verification server will include an `X-Signature`
+header. The value of this header will be the hex-encoded SHA-512 HMAC using the
+configured webhook secret as the HMAC secret.
+
+It is critical that your server validate the authenticity of the message. Here
+are some examples of validating the request payload:
+
+```go
+// secret is the webhook secret. It must be the same value as configured in the
+// verification server.
+const secret = "my-super-secret-value"
+
+func acceptPayload(w http.ResponseWriter, r *http.Request) {
+  defer r.Body.Close()
+
+  sig := w.Header.Get("X-Signature")
+  if sig == "" {
+    w.WriteHeader(400)
+    return
+  }
+
+  body, err := ioutil.ReadAll(r.Body)
```

Comment:
We should show max size read here instead of read all

Reply:
Done